root/etc/e-smith/templates/etc/ssh/sshd_config/70Restricted2Sftp

```diff
@@ -0,0 +1,27 @@
+{
+    # restrict groups to sftp
+
+    use strict;
+    use esmith::ConfigDB;
+    my $db = esmith::ConfigDB->open_ro();
+
+    my @SftpRestricted = ();
+    foreach ( split(',',$db->get_prop('sshd','AllowGroups') || '')) {
+        my ($group, $sftp) = split(':', $_);
+        if(!$group || !$sftp) {
+            next;
+        }
+        push @SftpRestricted, $group if ($sftp eq 'SftpRestricted');
+    }
+
+    if (@SftpRestricted) {
+        @SftpRestricted = join (',',@SftpRestricted);
```

Comment:
Assigning a scalar value to an array variable has an unpredictable semantic!

```suggestion
        my $matchGroup = join "," map { '"' . $_ . '"' } @SftpRestricted;
```

Reply:
code crash and service sshd broken, but thank I used your pseudo code to join a comma to the array